tests/e2e/uninstall.go

```diff
@@ -0,0 +1,194 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// requireRemoveKoperatorCRDs deletes the koperator CRDs
+func requireRemoveKoperatorCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing koperator CRDs", func() {
+		for _, crd := range koperatorCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
+		}
+	})
+}
+
+// requireUninstallingKoperator uninstall koperator Helm chart and removes Koperator's CRDs.
+func requireUninstallingKoperator(kubectlOptions k8s.KubectlOptions) {
+	When("Uninstalling Koperator", func() {
+		requireUninstallingKoperatorHelmChart(kubectlOptions)
+		requireRemoveKoperatorCRDs(kubectlOptions)
+	})
+}
+
+// requireUninstallingKoperatorHelmChart uninstall Koperator Helm chart
+// and checks the success of that operation.
+func requireUninstallingKoperatorHelmChart(kubectlOptions k8s.KubectlOptions) {
+	It("Uninstalling Koperator Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, koperatorLocalHelmDescriptor.ReleaseName, true)
```

Comment:
**Required**: We **MUST** check potential errors here.

```suggestion
		err := uninstallHelmChartIfExists(kubectlOptions, koperatorLocalHelmDescriptor.ReleaseName, true)
		
		Expect(err).NotTo(HaveOccurred())
```

Reply:
Fixed in [63ed2cd](https://github.com/banzaicloud/koperator/pull/987/commits/63ed2cdca59f233e329f912c042c39a21ecf3a7c)

Reply:
Fixed in [63ed2cd](https://github.com/banzaicloud/koperator/pull/987/commits/63ed2cdca59f233e329f912c042c39a21ecf3a7c)